src/MOI/MOI_callbacks.jl

```diff
@@ -51,114 +51,111 @@ function get_cb_solution(model::Optimizer, model_inner::XpressProblem)
     return
 end
 
-function applycuts(opt::Optimizer, model::XpressProblem)
-    itype = Cint(1)
-    interp = Cint(-1) # Get all cuts
-    delta = 0.0#Lib.XPRS_MINUSINFINITY
-    ncuts = Array{Cint}(undef, 1)
-    size = Cint(length(opt.cb_cut_data.cutptrs))
-    mcutptr = Array{Lib.XPRScut}(undef, size)
-    dviol = Array{Cdouble}(undef, size)
-    Lib.XPRSgetcpcutlist(
-        model,
-        itype,
-        interp,
-        delta,
-        ncuts,
+function _load_existing_cuts(model::Optimizer, cb_data::CallbackData)
+    if isempty(model.cb_cut_data.cutptrs)
+        return false
+    end
+    p_ncuts = Ref{Cint}(0)
+    size = length(model.cb_cut_data.cutptrs)
+    mcutptr = Vector{Lib.XPRScut}(undef, size)
+    dviol = Vector{Cdouble}(undef, size)
+    @checked Lib.XPRSgetcpcutlist(
+        cb_data.model,
+        1, # itype
+        -1, # interp
+        0.0, # delta
+        p_ncuts,
         size,
         mcutptr,
         dviol,
-    ) # requires an availabel solution
-    Lib.XPRSloadcuts(model, itype, interp, ncuts[1], mcutptr)
-    return ncuts[1] > 0
+    )
+    @checked Lib.XPRSloadcuts(cb_data.model, 1, -1, p_ncuts[], mcutptr)
+    return p_ncuts[] > 0
 end
 
 # ==============================================================================
 #    MOI callbacks
 # ==============================================================================
 
 function default_moi_callback(model::Optimizer)
-    return (cb_data) -> begin
+    function default_callback(cb_data)
+        # If we added a cut from the existing pool, it means that the current
+        # solution violates a cut that we previously added but that has since
+        # been deleted.
```

Comment:
I guess I don't really see why we need this. If Xpress has removed a cut, then we will at some point visit a violated solution and add a new one. The MOI callbacks do not enforce that once a solution is cut it will never be revisited.

And yeah, the `return` isn't tested, so I don't think this is actually doing anything.